src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -111,3 +112,73 @@ def post_execute():
 
         log.debug("Executing command post execution callback")
         flush_telemetry()
+
+
+@dataclasses.dataclass
+class SnowTyperCommandData:
+    func: Callable
+    args: Tuple[Any, ...]
+    kwargs: Dict[str, Any]
+
+
+class SnowTyper:
```

Comment:
Should we name it `SnowTyperFactory`?

Reply:
renamed:
`SnowTyper` -> `SnowTyperFactory`
`SnowTyperInstance` -> `SnowTyper`
`.create_app()` -> `.create_instance()`